independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -199,6 +199,10 @@ public ResourceClass createEndpoints(ClassInfo classInfo) {
         try {
             String path = scannedResourcePaths.get(classInfo.name());
             ResourceClass clazz = new ResourceClass();
+            if ((classInfo.enclosingClass() != null) && !Modifier.isStatic(classInfo.flags())) {
+                throw new DeploymentException("Resource class '" + classInfo.name()
+                        + "' must not have an enclosing class. Consider making this class static.");
```

Comment:
Not sure about the message. Maybe something like "Nested resource class '" + classInfo.name() "' has to be made static"?

Or `"Non static nested resources classes are not supported:" + classInfo.name()`.

Because being in an enclosing class is not an issue, right? The issue is not being static.

Reply:
Right. The message is indeed misleading

Reply:
Fixed